### PR TITLE
Logger replacement with Boost.Log2

### LINKED ISSUE #111: Check if we can substitute spdlog with Boost::Log2
Using spdlog for logging requires bringing a new dependency, of which we already have many. It may make sense to try substituting it with boost::log2, but the main concern here is - if the Boost::log2 is flexible enough to build advanced logging and profiling filters?

### PR DESCRIPTION
**Pull request by nemothenoone**
Resolves #111

### CODE REVIEW

**Comment by nkaskov** [comment]
All the merge commits have to be replaced with rebases in order to keep the commit history clean.

**Review by nkaskov — CHANGES_REQUESTED** [review]
All the merge commits have to be replaced with rebases in order to keep the commit history clean. Otherwise we have to much to review if something goes wrong.

**Comment by nemothenoone** [comment]
> We need to have a simple way to change logger level without changing logger's code.

There is a way to do this by interacting with Boost.Log2 core as the manual states explicitly in the second chapter:
https://www.boost.org/doc/libs/1_83_0/libs/log/doc/html/log/tutorial/trivial_filtering.html

File: include/nil/blueprint/asserts.hpp [hunk 1]
```diff
@@ -23,46 +23,14 @@
...
+#define CRYPTO3_ASSIGNER_ASSERTS_HPP
 
 #include <iostream>
 #include <cstring>
 
-#define UNREACHABLE(msg) ::nil::blueprint::unreachable((msg), __FILE__, __LINE__)
+#include <boost/assert.hpp>
 
-#define ASSERT(expr) ::nil::blueprint::assert_check((expr), #expr, __FILE__, __LINE__)
+#define BOOST_UNREACHABLE_MSG(msg) BOOST_VERIFY_MSG(false, "UNREACHABLE: "(msg))
```
[the remark this answers is not in the document]
Comment (nemothenoone) [reply include/nil/blueprint/asserts.hpp:34]: Hm. Not merging this in then. Will come back with some solution.

**Comment by nkaskov** [comment]
Logger functionality separated from asserts and moved to another PR: https://github.com/NilFoundation/zkllvm-assigner/pull/154

**Pull request closed without merging**